From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Mon, 4 Dec 2017 20:45:35 -0500
Subject: dash: Make sure item labels are only destroyed once

Labels are currently destroyed from both animateOutAndDestroy()
and destroy(), which now (rightfully) triggers a gjs warning. As
the label is created unconditionally since commit 36e5ae4a250,
mirror that and always release it in destroy() and hide it
elsewhere.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=791233
Origin: upstream, 3.26.3, commit:d5805f9f0df7bb6156179894b6b5f70e41261a52
---
 js/ui/dash.js | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/js/ui/dash.js b/js/ui/dash.js
index 0c89545..b89040b 100644
--- a/js/ui/dash.js
+++ b/js/ui/dash.js
@@ -178,15 +178,12 @@ var DashItemContainer = new Lang.Class({
     },
 
     destroy: function() {
-        if (this.label)
-            this.label.destroy();
-
+        this.label.destroy();
         this.parent();
     },
 
     animateOutAndDestroy: function() {
-        if (this.label)
-            this.label.destroy();
+        this.label.hide();
 
         if (this.child == null) {
             this.destroy();
